src/main/java/pl/wojciechkostecki/revel/service/LocalService.java

```diff
@@ -25,18 +25,19 @@ public Local save(Local local) {
 
     public Local findById(Long id) {
         return localRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException(String.format("Couldn't find a local with id: %s", id)));
+                .orElseThrow(() -> new EntityNotFoundException(String.format("Couldn't find a local with id: %d", id)));
     }
 
     public List<Local> findByName(String name) {
         return localRepository.findByNameContainingIgnoreCase(name);
     }
 
     public Local updateLocal(Long id, Local local) {
-        if(id == null){
-            throw new NullPointerException(String.format("No local with the given id %s", id));
+        if (local.getId().equals(id)) {
```

Comment:
Let's stop comparing these ids. Also your code might throw NPE now :D. For now our update will be really simple. Just saving.

Reply:
I wonder that if I just save it, a completely new object may be created, not an existing one will not be updated.  Now see the changes, and if I still make mistakes, I will only set saving 😃